libs/transloco/src/lib/transloco.transpiler.ts

```diff
@@ -16,7 +16,7 @@ import {
 export const TRANSLOCO_TRANSPILER = new InjectionToken('TRANSLOCO_TRANSPILER');
 
 export interface TranslocoTranspiler {
-  transpile(value: any, params: HashMap, translation: Translation): any;
+  transpile(value: any, params: HashMap, translation: Translation, key?: any): any;
```

Comment:
Why is this optional and typed as any? Please add the following comment above:
```
// TODO Change parameters to object in the next major release
```

Reply:
Added TODO.

I added `key` as optional since the built in transpilers don't consume it, and it allowed me to change less of the core library. However after thinking about it more I understand better how the transpilers are hooked up, so I have removed the optional type, and am now passing in 'key' wherever `transpile` is being called.

This does mean that I had to update the specs to now pass in a key. I have not added any specs since because this isn't really adding any functionality to the built in transpilers, i'm not sure what exactly I should be testing. I will definitely need help understanding what I should be looking for in the specs.

As for the `any` type, I was following the type of the `value` parameter. Looking at the `translocoService.translate` method, `key` should be `TranslateParams` correct?

Reply:
@jacksonopp I don't think we need to add any specs here, it's just passing another param.
The key type is `string`